syntax = "proto3";
package protoconf;
option go_package = "github.com/tableauio/loader/test/protoconf";

import "tableau/protobuf/tableau.proto";
import "common_conf.proto";

option (tableau.workbook) = {
  name: "Test.xlsx"
};

message ActivityConf {
  option (tableau.worksheet) = {
    name: "Activity"
    ordered_map: true
  };

  map<uint64, Activity> activity_map = 1 [(tableau.field) = { key: "ActivityID" layout: LAYOUT_VERTICAL}];
  message Activity {
    uint64 activity_id = 1 [(tableau.field) = { name: "ActivityID"}];
    string activity_name = 2 [(tableau.field) = { name: "ActivityName" }];
    map<uint32, Chapter> chapter_map = 3 [(tableau.field) = { key: "ChapterID" layout: LAYOUT_VERTICAL }];
    message Chapter {
      uint32 chapter_id = 1 [(tableau.field) = { name: "ChapterID"  prop:{refer:"ItemConf.ID"}}];
      string chapter_name = 2 [(tableau.field) = { name: "ChapterName" }];
      map<uint32, Section> section_map = 3 [(tableau.field) = { key: "SectionID" layout: LAYOUT_VERTICAL }];
    //   message Section {
    //     uint32 section_id = 1 [(tableau.field) = { name: "SectionID" }];
    //     string section_name = 2 [(tableau.field) = { name: "SectionName" }];
    //     repeated Item section_item_list = 3 [(tableau.field) = { name: "SectionItem" layout: LAYOUT_HORIZONTAL }];
    //     message Item {
    //       int32 id = 1 [(tableau.field) = { name: "Id" }];
    //       int32 num = 2 [(tableau.field) = { name: "Num" }];
    //     }
    //   }
    }
  }
  map<uint32, Item> section_item_map = 2 [(tableau.field) = { key: "Id" layout: LAYOUT_VERTICAL }];
}

message Section {
  uint32 section_id = 1 [(tableau.field) = { name: "SectionID" }];
  string section_name = 2 [(tableau.field) = { name: "SectionName" }];
  repeated Item section_item_list = 3 [(tableau.field) = { name: "SectionItem" layout: LAYOUT_HORIZONTAL }];
  map<uint32, int32> section_rank_map = 4 [(tableau.field) = {layout: LAYOUT_VERTICAL }];
}